bin/compile

```diff
@@ -291,9 +292,23 @@ prune_devdependencies() {
   fi
 }
 
-meta_set "build-step" "prune-dependencies"
-header "Pruning devDependencies" | output "$LOG_FILE"
-prune_devdependencies | output "$LOG_FILE"
+# if we are using yarn
+# and in the new feature
+# and we're using the default cache directories
+# then save off the cache after we prune out devDepenencies
+if [[ "$YARN" == "true" ]] || [[ $(features_get "cache-native-yarn-cache") == "true" ]] || [[ "$(get_cache_directories "$BUILD_DIR")" == "" ]]; then
```

Comment:
Note: I suspect that I will rip out the caching system and re-do it once we have results that can inform us as to how it should function in 2019. This kind of switch isn't elegant code, but is the quickest way to gather that data :)